### PR TITLE
(moveit_py) Extend Trajectory Execution Manager

### PR DESCRIPTION
**Pull request by JensVanhooydonck**
Added part of the functions from #2442

### Description

Added execute, execute_and_wait, wait_for_execution and get_last_execution_status.
I also added py::call_guard<py::gil_scoped_release>() for the (possible) long running calls to free up the ptyhon thread.

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit2/blob/main/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://moveit.picknik.ai/humble/doc/examples/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit2/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]

File: moveit_py/moveit/planning.pyi [hunk 1]
```diff
@@ -79,6 +79,10 @@ class TrajectoryExecutionManager:
     def is_managing_controllers(self, *args, **kwargs) -> Any: ...
     def process_event(self, *args, **kwargs) -> Any: ...
     def push(self, *args, **kwargs) -> Any: ...
+    def execute(self, *args, **kwargs) -> Any: ...
```
[the remark this answers is not in the document]
Comment (JensVanhooydonck) [reply moveit_py/moveit/planning.pyi]: Stubgen didn't work for me.. (unable to import) Always tried it in the moveit2/moveit_py directory. Fixed this by running in main directory.

File: moveit_py/src/moveit/moveit_ros/trajectory_execution_manager/trajectory_execution_manager.cpp [hunk 2]
```diff
@@ -157,14 +157,32 @@ void initTrajectoryExecutionManager(py::module& m)
       // ToDo(MatthijsBurgh)
       // See https://github.com/ros-planning/moveit2/issues/2442
       // get_trajectories
-      // execute
-      // execute_and_wait
-      // wait_for_execution
       // get_current_expected_trajectory_index
```
[the remark this answers is not in the document]
Comment (JensVanhooydonck) [reply moveit_py/src/moveit/moveit_ros/trajectory_execution_manager/trajectory_execution_manager.cpp]: Added extra comment block

File: moveit_py/src/moveit/moveit_ros/trajectory_execution_manager/trajectory_execution_manager.cpp [hunk 3]
```diff
@@ -157,14 +157,32 @@ void initTrajectoryExecutionManager(py::module& m)
...
-      // get_last_execution_status
-
+      .def("execute",
+           py::overload_cast<const trajectory_execution_manager::TrajectoryExecutionManager::ExecutionCompleteCallback&,
+                             bool>(&trajectory_execution_manager::TrajectoryExecutionManager::execute),
+           py::arg("callback"), py::arg("auto_clear") = true,
+           R"(
+           Execute a trajectory.
```
[the remark this answers is not in the document]
Comment (JensVanhooydonck) [reply moveit_py/src/moveit/moveit_ros/trajectory_execution_manager/trajectory_execution_manager.cpp]: Altered the docstring of the execute commands.

File: moveit_py/src/moveit/moveit_ros/trajectory_execution_manager/trajectory_execution_manager.cpp [hunk 4]
```diff
@@ -157,14 +157,32 @@ void initTrajectoryExecutionManager(py::module& m)
...
+           )")
+      .def("execute_and_wait", &trajectory_execution_manager::TrajectoryExecutionManager::executeAndWait,
+           py::call_guard<py::gil_scoped_release>(), py::arg("auto_clear") = true,
```
[the remark this answers is not in the document]
Comment (JensVanhooydonck) [reply moveit_py/src/moveit/moveit_ros/trajectory_execution_manager/trajectory_execution_manager.cpp]: Good catch. Moved

**Comment by MatthijsBurgh** [comment]
@JensVanhooydonck You have some whitespace issues in `trajectory_execution_manager.cpp. See the formatting job.
`

**Review by MatthijsBurgh — APPROVED** [review]

**Review by sjahr — APPROVED** [review]

**Pull request merged**